core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -70,6 +69,8 @@ public static XContentBuilder builder(XContent xContent, Set<String> includes, S
 
     private boolean humanReadable = false;
 
+    private TimeUnit timeUnit = TimeUnit.MILLISECONDS;
```

Comment:
I see that differently. I like the idea to allow unit to be configurable and I think it helps us to be more coherent in field names as well as better identify the places where we expect a given time unit because of further parsing.